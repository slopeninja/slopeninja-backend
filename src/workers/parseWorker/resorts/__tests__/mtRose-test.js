import fs from 'fs';
import {
  parseMtRoseSnow,
  parseMtRoseLiftCounts,
  parseMtRoseTrailCounts,
  parseMtRoseLifts,
  parseMtRoseTrails,
} from '../mtRose';
import { createHtmlParser } from '../../parserFactory';

test('fetches MtRose snow data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/mtRose-weather.html.input`);

  const resortData = await createHtmlParser('snow', parseMtRoseSnow)(htmlText);
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: 'snow',
      temperature: 26,
      baseCondition: null,
      newSnow: 12,
      snowDepthBase: 240,
      snowDepthSummit: 302,
    },
  });
});

test('fetches all null for nonexisting snow values', async () => {
  const resortData = await createHtmlParser('snow', parseMtRoseSnow)('<html></html>');
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: null,
      snowDepthBase: null,
      snowDepthSummit: null,
    },
  });
});

test('fetches MtRose lifts data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/mtRose-weather.html.input`);
  const resortData = await createHtmlParser('liftCounts', parseMtRoseLiftCounts)(htmlText);
  expect(resortData).toEqual({
    liftCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches all null for nonexisting lift values', async () => {
  const resortData = await createHtmlParser('liftCounts', parseMtRoseLiftCounts)('<html></html>');
  expect(resortData).toEqual({
    liftCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches MtRose trails data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/mtRose-weather.html.input`);
  const resortData = await createHtmlParser('trailCounts', parseMtRoseTrailCounts)(htmlText);
  expect(resortData).toEqual({
    trailCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches all null for nonexisting trails values', async () => {
  const resortData = await createHtmlParser('trailCounts', parseMtRoseTrailCounts)('<html></html>');
  expect(resortData).toEqual({
    trailCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches MtRose lift list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/mtRose-weather.html.input`);
  const resortData = await createHtmlParser('lifts', parseMtRoseLifts)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('lifts', parseMtRoseLifts)('<html></html>');
  expect(resortData).toMatchObject({ lifts: [] });
});

test('fetches MtRose trail list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/mtRose-weather.html.input`);
  const resortData = await createHtmlParser('trails', parseMtRoseTrails)(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser('trails', parseMtRoseTrails)('<html></html>');
  expect(resortData).toMatchObject({ trails: [] });
});
