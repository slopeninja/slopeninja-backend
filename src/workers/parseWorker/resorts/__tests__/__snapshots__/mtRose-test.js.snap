// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches MtRose lift list correctly 1`] = `
Object {
  "lifts": Array [
    Object {
      "category": null,
      "name": "Northwest Express",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Lakeview Express",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Zephyr Express",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Wizard",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Galena",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Magic East",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Magic West",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Chuter",
      "status": "closed",
    },
  ],
}
`;

exports[`fetches MtRose trail list correctly 1`] = `
Object {
  "trails": Array [
    Object {
      "category": "Main Side",
      "level": "advanced",
      "name": "AIDA’S",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "advanced",
      "name": "BULLWHIP",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "advanced",
      "name": "EXPRESS",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "advanced",
      "name": "GREG’S GAMBLE",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "advanced",
      "name": "JETTA",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "advanced",
      "name": "NORTHWEST PASSAGE",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "advanced",
      "name": "SIX GUN",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "advanced",
      "name": "SUNSET",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "advanced",
      "name": "TAHOE VISTA",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "advanced",
      "name": "WATERFALL",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "intermediate",
      "name": "KIT CARSON BOWL",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "intermediate",
      "name": "KIT CARSON TRAVERSE",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "intermediate",
      "name": "BUNKER",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "intermediate",
      "name": "LONE TREE",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "intermediate",
      "name": "PONDEROSA",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "intermediate",
      "name": "RAMSEY’S",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "intermediate",
      "name": "UPPER NORTHWEST",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "ADVENTURE TRAIL",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "AROUND THE WORLD",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "BROOMSTICK",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "BUCKBOARD",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "DRAGON’S TAIL",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "EASY STREET",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "GALENA",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "LARIAT",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "LAKESIDE TRAIL",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "MERLIN",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "MAGIC POTION",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "MYSTIC",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "NORTH RIM",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "OLD SCHOOL",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "RELAY",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "SHOW OFF",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "SPELLBOUND",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "WARLOCK",
      "status": "open",
    },
    Object {
      "category": "Main Side",
      "level": "beginner",
      "name": "WITCHES BREW",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "advanced",
      "name": "CENTRAL PACIFIC",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "advanced",
      "name": "GOLD RUN",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "advanced",
      "name": "OUTLAW",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "advanced",
      "name": "WASHOE ZEPHYR",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "advanced",
      "name": "WILD CARD",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "intermediate",
      "name": "BADLANDS",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "intermediate",
      "name": "BIG BONANZA",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "intermediate",
      "name": "BONANZA TRAVERSE",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "intermediate",
      "name": "BRUCE’S",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "intermediate",
      "name": "HIGH TRAVERSE",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "intermediate",
      "name": "LOG FLUME",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "intermediate",
      "name": "MINE TRAIN",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "intermediate",
      "name": "SILVER DOLLAR",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "intermediate",
      "name": "SLIDE BOWL",
      "status": "open",
    },
    Object {
      "category": "Slide Bowl",
      "level": "intermediate",
      "name": "SUNRISE BOWL",
      "status": "open",
    },
  ],
}
`;
